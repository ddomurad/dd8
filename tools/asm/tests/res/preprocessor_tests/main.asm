.def (
  ZPAGE := 0x0000
  ENTRY := 0xe000
  LIB_CODE := 0xf000

  ARGS := <0x0000, 0x0001, 0x0002> ; function call args
)

.tmpl call_func (fname, args) {
  .rep i, 0, args.len-1 {
    ldai args[i]
    sta ARGS[i]
  }
  jsr fname
}

.org ZPAGE
; ARG_0: .byte
; ARG_1: .byte
; ARG_2: .byte
; ARG_3: .byte


.org ENTRY
@call_func (test_function, <0x10, 0x20>)

.org LIB_CODE

test_function: {
  ldx ARGS[0] 
  ldy ARGS[1]
  rts
}
