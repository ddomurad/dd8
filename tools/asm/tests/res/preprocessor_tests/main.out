:0BE00000A9108500A92085012000F078
:05F00000A600A4016060
:00000001FF
